File: Others/easy/NodeDepths.py
def node_depths(root, curr_depth):
    if not root:
        return 0

    return curr_depth + node_depths(root.left, curr_depth + 1) + node_depths(root.right, curr_depth + 1)


def nodeDepths(root):
    return node_depths(root, 0)


# This is the class of the input binary tree.
class BinaryTree:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

File: Golang/easy/node-depths.go
package main

type BinaryTree struct {
	Value       int
	Left, Right *BinaryTree
}

func Solve(root *BinaryTree, depth int) int {
	if root == nil {
		return 0
	}

	leftRes := Solve(root.Left, depth+1)
	rightRes := Solve(root.Right, depth+1)

	return depth + leftRes + rightRes
}

func NodeDepths(root *BinaryTree) int {
	return Solve(root, 0)
}
